jhove-apps/src/main/java/edu/harvard/hul/ois/jhove/viewer/RepTreeRoot.java

```diff
@@ -438,11 +439,25 @@ private void addArrayMembers(DefaultMutableTreeNode node, Property p) {
 		if (null != typ)
 			switch (typ) {
 			case INTEGER: {
-				addToNode(node, (Integer[]) pVal);
+				if (pVal instanceof int[]) {
+					Integer[] vals = Arrays.stream((int[])pVal) // IntStream
+							.boxed()				// Stream<Integer>
+							.toArray(Integer[]::new);
+					addToNode(node, vals);
+				} else {
+					addToNode(node, (Integer[]) pVal);
+				}
 				break;
 			}
 			case LONG: {
-				addToNode(node, (Long[]) pVal);
+				if (pVal instanceof long[]) {
+					Long[] vals = Arrays.stream((long[])pVal) // IntStream
+							.boxed()				// Stream<Integer>
+							.toArray(Long[]::new);
+					addToNode(node, vals);
+				} else {
+					addToNode(node, (Long[]) pVal);
+				}
```

Comment:
Might the remaining cases not also be subject to this issue, e.g. arrays of float or double primitives?